src/flb_pipe.c

```diff
@@ -182,3 +182,30 @@ ssize_t flb_pipe_write_all(int fd, const void *buf, size_t count)
 
     return total;
 }
+
+/* Writes to a non-blocking pipe yielding if no more bytes can be written */
+ssize_t flb_pipe_write_async(int fd, const void *buf, size_t count, struct flb_thread *th)
+{
+    ssize_t bytes;
+    size_t total = 0;
+
+    do {
+        bytes = flb_pipe_w(fd, (const char *) buf + total, count - total);
+        if (bytes == -1) {
+            if (FLB_PIPE_WOULDBLOCK()) {
+                flb_thread_yield(th, FLB_FALSE);
```

Comment:
I am thinking about this approach....

If the pipe is full and might block, if we yield, when are we going to try to send the return value again over the pipe ?, I mean, what's the event loop condition to resume on this point ?

Reply:
That's a good point. My bad.
Indeed in this case we simply hiding the problem and the coroutine in question will never be able to run to completion.

The reason we haven't noticed it is probably because it happens after output flush completes and it doesn't happen often enough.

I'll try to fix this behavior it seems that I need to add `mk_event_add` prior to yielding and `mk_event_del` after that. But in this case I'll need to pass an event loop instance (not a problem) and create an event.

Reply:
@edsiper thank you for pointing to my mistake. I make some changes and checked that thread is indeed resumed.

Please take a look at proposed solution when you have time.

Reply:
@edsiper can you please help here?